cfgov/v1/handlers/blocks/feedback.py

```diff
@@ -59,22 +59,37 @@ def get_response(self, form):
 
     def success(self):
         if self.request.is_ajax():
-            return JsonResponse({'result': 'pass'})
+            if get_feedback_type(self.block_value) == 'suggestion':
+                return JsonResponse(
+                    {'result': 'pass',
+                     'heading': 'Thank you!',
+                     'message': "Be sure to also sign up for our email list "
+                     "to get our blog posts and other tips about homebuying "
+                     "and mortgages in your inbox. We'll also let you know "
+                     "when we make updates to Owning a Home."}
+                )
+            else:
+                return JsonResponse(
+                    {'result': 'pass',
+                     'heading': '',
+                     'message': 'Thanks for your feedback!'}
+                )
         else:
             messages.success(self.request, message='Thanks for your feedback!')
             return HttpResponseRedirect(self.page.url)
 
     def fail(self, form):
-        if self.request.is_ajax():
-            return JsonResponse({'result': 'fail'})
-
+        if form.errors.get('is_helpful', None):
+            msg = 'You must select an option.'
+        elif form.errors.get('comment', None):
+            msg = 'You must enter a comment.'
+        elif form.errors.get('email', None):
+            msg = 'You must enter a valid email.'
         else:
-            if form.errors.get('is_helpful', None):
-                messages.error(self.request, 'You must select an option.')
-            elif form.errors.get('comment', None):
-                messages.error(self.request, 'You must enter a comment.')
-            else:
-                message = 'Something went wrong. Please try again.'
-                messages.error(self.request, message)
+            msg = 'Something went wrong. Please try again.'
 
-        return {'form': form}
+        if self.request.is_ajax():
+            return JsonResponse({'result': 'fail', 'message': msg})
+        else:
+            messages.error(self.request, msg)
+            return {'form': form}
```

Comment:
If you come to the feedback page with `?is_helpful=1` set, and then try to submit the form without entering a message, it re-renders the page but now you've lost the query string (because it POSTS back to the same URL). Is there some way of setting the field value directly on the form?